packages/verified-fetch/test/utils/parse-url-string.spec.ts

```diff
@@ -309,15 +270,42 @@ describe('parseUrlString', () => {
     })
 
     it('can parse a URL with PeerId+path+queryString', async () => {
+      ipns.resolve.withArgs(matchPeerId(testPeerId)).resolves({
+        cid: CID.parse('QmQJ8fxavY54CUsxMSx9aE9Rdcmvhx8awJK2jzJp4iAqCr'),
+        path: ''
+      })
       const result = await parseUrlString({
-        urlString: `ipns://${testPeerId.toString()}/some/path/to/file.txt?fomat=dag-cbor`,
+        urlString: `ipns://${testPeerId}/some/path/to/file.txt?fomat=dag-cbor`,
         ipns,
-        logger: defaultLogger()
+        logger
       })
       expect(result.protocol).to.equal('ipns')
       expect(result.cid.toString()).to.equal('QmQJ8fxavY54CUsxMSx9aE9Rdcmvhx8awJK2jzJp4iAqCr')
       expect(result.path).to.equal('some/path/to/file.txt')
       expect(result.query).to.deep.equal({ fomat: 'dag-cbor' })
     })
+
+    it('should parse an ipns:// url with a path that resolves to a CID with a path', async () => {
+      const cid = CID.parse('QmQJ8fxavY54CUsxMSx9aE9Rdcmvhx8awJK2jzJp4iAqCr')
+      const peerId = await createEd25519PeerId()
+      const recordPath = 'foo'
```

Comment:
Do you think it might make sense to add another test with:

      const recordPath = 'foo/'

To ensure there's no funky business with the slashes?

Reply:
Good shout, I've added some more tests

Reply:
Nice!